topologies/multi.go

```diff
@@ -90,60 +97,43 @@ func (t *MultiTopology) GenerateBGPPeers(routeReflectors []corev1.Node, nodes ma
 
 	peers := int(math.Min(float64(len(routeReflectors)), 3))
 
-	rrIndex := -1
-	rrIndexPerZone := map[string]int{}
+	isMultiZone := t.IsMultiZone(nodes)
 
+	rand.Seed(time.Now().UnixNano())
 	for n := range nodes {
 		if t.IsRouteReflector(string(n.GetUID()), n.GetLabels()) {
 			continue
 		}
 
-		routeReflectorsForNode := []corev1.Node{}
-
-		if t.Config.ZoneLabel != "" {
-			nodeZone := n.GetLabels()[t.Config.ZoneLabel]
-			rrsSameZone := []corev1.Node{}
+		routeReflectorsForNode := map[*corev1.Node]bool{}
 
-			for i, rr := range routeReflectors {
-				rrZone := rr.GetLabels()[t.Config.ZoneLabel]
-				if nodeZone == rrZone {
-					if _, ok := rrIndexPerZone[nodeZone]; !ok {
-						rrIndexPerZone[nodeZone] = -1
-					}
+		// Select the first two RRs from different zones in Multi Zone clusters
+		if isMultiZone {
+			firstRR := &routeReflectors[rand.Intn(len(routeReflectors))]
```

Comment:
The RR selector must be deterministic otherwise it will rebalance the cluster on each reconcile cycle.